### PR TITLE
feat: support user delete in the rest API

### PR DESCRIPTION
**Pull request by Ben-Sheppard**
## Description

This PR introduces the endpoint to handle user deletion as per the spec [here](https://docs.google.com/document/d/1rS989EESojWRrCMLH7Nscy1UppbgOanvzVeOljCc_iA/edit?tab=t.0#heading=h.m5jo0jd7jm0d)

### CODE REVIEW

**Review by pepopowitz — APPROVED** [review]
spec changes look good to me 👍

File: zeebe/gateway-protocol/src/main/proto/rest-api.yaml [hunk 1]
```diff
@@ -2421,6 +2370,57 @@ paths:
...
+          content:
+            application/problem+json:
+              schema:
+                $ref: "#/components/schemas/ProblemDetail"
+        "500":
+          description: An internal error occurred while processing the request.
+          content:
+            application/problem+json:
+              schema:
+                $ref: "#/components/schemas/ProblemDetail"
```
Comment (emulbreh) [inline zeebe/gateway-protocol/src/main/proto/rest-api.yaml:2422]:
```suggestion
          $ref: "#/components/responses/InternalServerError"
```

File: zeebe/gateway-protocol/src/main/proto/rest-api.yaml [hunk 2]
```diff
@@ -2421,6 +2370,57 @@ paths:
...
+              schema:
+                $ref: "#/components/schemas/UserCreateResponse"
+        "400":
+          description: |
+            The user could not be created.
+            More details are provided in the response body.
+          content:
+            application/problem+json:
+              schema:
+                $ref: "#/components/schemas/ProblemDetail"
```
Comment (emulbreh) [inline zeebe/gateway-protocol/src/main/proto/rest-api.yaml:2400]:
```suggestion
          $ref: "#/components/responses/Unauthorized"
```

Comment (emulbreh) [reply zeebe/gateway-protocol/src/main/proto/rest-api.yaml:2400]: I just added these common responses in https://github.com/camunda/camunda/pull/24546

**Review by emulbreh — APPROVED** [review]

File: zeebe/gateway-protocol/src/main/proto/rest-api.yaml [hunk 3]
```diff
@@ -2275,7 +2281,7 @@ paths:
                 $ref: "#/components/schemas/AuthorizationSearchResponse"
         "400":
           description: >
-            The user authorization search query failed.
+            The user delete failed.
```
Comment (tmetzke) [inline zeebe/gateway-protocol/src/main/proto/rest-api.yaml:2284]:
```suggestion
            The user authorization search query failed.
```

File: zeebe/gateway-protocol/src/main/proto/rest-api.yaml [hunk 4]
```diff
@@ -2200,6 +2159,53 @@ paths:
...
+              schema:
+                $ref: "#/components/schemas/ProblemDetail"
+        "401":
+          description: |
+            The request to create a user was unauthorized.
+            More details are provided in the response body.
+          content:
+            application/problem+json:
+              schema:
+                $ref: "#/components/schemas/ProblemDetail"
```
Comment (tmetzke) [inline zeebe/gateway-protocol/src/main/proto/rest-api.yaml:2197]:
```suggestion
        "401":
          $ref: "#/components/responses/Unauthorized"
```

File: zeebe/gateway-protocol/src/main/proto/rest-api.yaml [hunk 5]
```diff
@@ -2200,6 +2159,53 @@ paths:
...
+      responses:
+        "202":
+          description: |
+            The user was created successfully.
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/UserCreateResponse"
+        "400":
+          description: >
```
Comment (tmetzke) [inline zeebe/gateway-protocol/src/main/proto/rest-api.yaml:2183]:
```suggestion
          description: |
```

**Review by tmetzke — APPROVED** [review]
👍 Looks correct now, thanks @Ben-Sheppard 
🔧 Some minor improvement suggestions, nothing blocking. Pre-approving already.

**Pull request merged**